plugins/module_utils/network/sonic/config/interfaces/interfaces.py

```diff
@@ -174,7 +175,7 @@ def _state_replaced(self, want, have, diff):
         """
         commands = self.filter_comands_to_change(diff, have)
         requests = self.get_delete_interface_requests(commands, have)
-        requests.extend(self.get_modify_interface_requests(commands, have))
+        requests.extend(self.get_modify_interface_requests(want, have))
```

Comment:
Still if want is subset of have, then diff and commands are empty, so that there is nothing to be deleted and replaced. if the config for an interface in want is subset of those in have, the config for the interface in want should replace the config for same interface in have. But the code does nothing in such case.